ui/lib/feed/src/components/beam-feed.tsx

```diff
@@ -51,8 +53,47 @@ const BeamFeed = (props: BeamFeedProps) => {
     offsetTop,
   } = props;
 
+  const { showNsfw } = useNsfwToggling();
+  const { data: loginData, loading: authenticating } = useGetLogin();
+  const isLoggedIn = !!loginData?.id;
+
+  let nsfwFilters;
+
+  /**
```

Comment:
Is it possible to extract all this into a local function something `getNsfwFilter` ? Otherwise its getting harder to read this beam-feed file.